tests/parser/features/test_external_contract_calls.py

```diff
@@ -199,6 +199,28 @@ def __init__(arg1: address):
     print('Successfully executed a multiple external contract calls')
 
 
+def test_invalid_external_contract_call_to_the_same_contract(assert_tx_failed):
+    contract = """
+class Bar():
+    def bar() -> num: pass
+
+def bar() -> num:
+    return 1
+
+def _expr(x: address) -> num:
+    return Bar(x).bar()
+
+def _stmt(x: address) -> num:
+    return Bar(x).bar()
+    """
+    t.s = t.Chain()
+    c = get_contract(contract)
+    c._expr(t.a1)
```

Comment:
This should fail because we passed an account address instead of a contract address IMO.
Maybe insert `assert x.is_contract` in the underlying module for using external contracts?

Reply:
I agree with you, this pr is just to make sure external contract calls to the same contract don't work, I'll make another one that throws if there's no contract.